_posts/2022-05-12-data-pipelines-in-Databricks.markdown

```diff
@@ -0,0 +1,240 @@
+---
+layout: post
+title: "Data pipelines in Databricks"
+date: 2022-05-12
+tags: Databricks data testing
+author: Natalie Laing
+image: /img/code.jpg
+---
+
+# Data pipelines in Databricks
+
+## Introduction
+
+![delta lake](/img/blog/databricks-data-pipeline/delta-lake.png)
+<br />
+
+### Delta Lake
+
+What is a delta lake?
+
+Delta lake enables developers to create Lakehouse architectures on top of data lakes.
+
+Delta lake is open source.
+Delta lakehouse combines elements from a data warehouse and a data lake.
+
+![data lakehouse](/img/blog/databricks-data-pipeline/data-lakehouse.png)
+
+### Databricks
+
+What is Databricks?
+
+Databricks provides a solution for ingestion, processing and scheduling allowing developers to run ETL workloads directly on a data lake.
+
+What is a data pipeline in Databricks?
+
+A data pipeline is only concerned with the transformation of Databricks job code, this usually consists of moving data from bronze to gold classification.
+
+- Bronze
+  - Ingest raw data
+- Silver
+  - Deduplicate data
+  - Enforce schemas
+- Gold
+  - Perform business-specific transforms to data
+  - Aggregate data
+
+<br/>
+
+### Multi task jobs v delta live jobs
+
+<br/>
+
+Multi-task jobs:
+
+- This is suited for batch jobs. Multi task jobs need to wait for a task to finish before it can move onto the next task. If you have a streaming job the multi task job cannot determine when to finish a task and move onto the next.
+
+- Allows you to execute notebooks in a flow
+
+This allows you to set your bronze - silver layer dependencies and run the pipeline in that set sequence.
+This allows you to have a single workflow or parallel workflows so you can build your data pipeline in a way that suits your individual needs.
+
+Databricks enables you to run your data pipeline on a schedule that is as simple as setting up a cron job in your IaC. The alternative is to go into the Databricks UI and manually trigger your data pipeline.
+
+Delta live:
+
+- This is suited for streaming jobs.
+- More suited for creating ETL pipelines.
+
+Although delta live would have been better suited for our needs we did not use delta live as it was still in preview and not ready for production workflows at the time.
+
+## Challenges
+
+What challenges did we face?
```

Comment:
So far the blog post reads as a lot of general definitions then now it goes into a personal language as what did we face and how did we do it. 
Maybe try thinking about how you can connect the previous sections with getting into challenges and how to sections